docs/source/cpp/compute.rst

```diff
@@ -0,0 +1,419 @@
++==========================+============+====================+=====================+
+| add                      | Binary     | Numeric            | Numeric             |
++--------------------------+------------+--------------------+---------------------+
+| add_checked              | Binary     | Numeric            | Numeric             |
++--------------------------+------------+--------------------+---------------------+
+| multiply                 | Binary     | Numeric            | Numeric             |
++--------------------------+------------+--------------------+---------------------+
+| multiply_checked         | Binary     | Numeric            | Numeric             |
++--------------------------+------------+--------------------+---------------------+
+| subtract                 | Binary     | Numeric            | Numeric             |
++--------------------------+------------+--------------------+---------------------+
+| subtract_checked         | Binary     | Numeric            | Numeric             |
++--------------------------+------------+--------------------+---------------------+
+
+Comparisons
+~~~~~~~~~~~
+
+Those functions expect two inputs of the same type and apply a given
+comparison operator.  If any of the input elements in a pair is null,
+the corresponding output element is null.
+
++--------------------------+------------+---------------------------------+---------------------+
+| Function names           | Arity      | Input types                     | Output type         |
++==========================+============+=================================+=====================+
+| equal, not_equal         | Binary     | Numeric                         | Boolean             |
++--------------------------+------------+---------------------------------+---------------------+
+| equal, not_equal         | Binary     | Binary- and String-like         | Boolean             |
++--------------------------+------------+---------------------------------+---------------------+
+| equal, not_equal         | Binary     | Temporal                        | Boolean             |
++--------------------------+------------+---------------------------------+---------------------+
+| greater, greater_equal,  | Binary     | Numeric                         | Boolean             |
+| less, less_equal         |            |                                 |                     |
++--------------------------+------------+---------------------------------+---------------------+
+| greater, greater_equal,  | Binary     | Binary- and String-like         | Boolean             |
+| less, less_equal         |            |                                 |                     |
++--------------------------+------------+---------------------------------+---------------------+
+| greater, greater_equal,  | Binary     | Temporal                        | Boolean             |
```

Comment:
Since this doc uses them throughout, it would be good to have an explicit list somewhere of what is included in "numeric", "binary- and string-like", "temporal", and any other type categories used.

Reply:
Will do, though unfortunately the real answer may be slightly function-specific.